src/Dock.Avalonia.Themes.Default/PinHostControl.xaml

```diff
@@ -0,0 +1,107 @@
+﻿<Styles xmlns="https://github.com/avaloniaui"
+        xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+        xmlns:id="clr-namespace:Dock.Avalonia;assembly=Dock.Avalonia"
+        xmlns:idc="clr-namespace:Dock.Avalonia.Controls;assembly=Dock.Avalonia"
+        xmlns:converters="clr-namespace:Dock.Avalonia.Converters;assembly=Dock.Avalonia"
+        xmlns:Dock="clr-namespace:Dock.Model;assembly=Dock.Model"
+        xmlns:Controls="clr-namespace:Dock.Model.Controls;assembly=Dock.Model">
+
+    <Style Selector="idc|PinHostControl">
+        <Style.Resources>
+            <converters:PinDockToGridLengthConverter x:Key="PinDockToGridLengthConverter"/>
+        </Style.Resources>
+        <Setter Property="Template">
+            <ControlTemplate>
+                <DockPanel Background="Transparent">
+                    <ContentControl Content="{Binding Left}" DockPanel.Dock="Left"/>
+                    <ContentControl Content="{Binding Right}" DockPanel.Dock="Right"/>
+                    <Grid>
+                        <Panel x:Name="LeftPanel"
+                               HorizontalAlignment="Left" DataContext="{Binding Left}" IsVisible="{Binding IsExpanded, FallbackValue=False}"
+                               Background="White" ZIndex="100">
+                            <idc:DockToolChrome IsActive="{Binding IsActive}" Content="{Binding ActiveDockable}"/>
+                        </Panel>
+                        <Panel x:Name="RightPanel"
+                               HorizontalAlignment="Right" DataContext="{Binding Right}" IsVisible="{Binding IsExpanded, FallbackValue=False}"
+                               Background="White" ZIndex="100">
+                            <idc:DockToolChrome IsActive="{Binding IsActive}" Content="{Binding ActiveDockable}"/>
+                        </Panel>
+                        <Grid Background="Transparent">
+                            <Grid.ColumnDefinitions>
```

Comment:
These columns are here just to shrink center area when AutoHide = false.
I've made it so vertical columns have priority, like visual studio.